documents/coredocument.go

```diff
@@ -634,41 +634,41 @@ func (cd *CoreDocument) Timestamp() (time.Time, error) {
 }
 
 // AddAttribute adds a custom attribute to the model with the given value. If an attribute with the given name already exists, it's updated.
-func (cd *CoreDocument) AddAttribute(name string, attributeType attributeType, value string) (*CoreDocument, error) {
+func (cd *CoreDocument) AddAttribute(keyLabel string, attributeType attributeType, value string) (*CoreDocument, error) {
 	ncd, err := cd.PrepareNewVersion(nil, CollaboratorsAccess{})
 	if err != nil {
 		return nil, errors.NewTypedError(ErrCDAttribute, errors.New("failed to prepare new version: %v", err))
 	}
 	// TODO convert value from string to correct type
 	// For now its only string that is supported
-	nAttr, err := newAttribute(name, attributeType, value)
+	nAttr, err := newAttribute(keyLabel, attributeType, value)
 	if err != nil {
 		return nil, err
 	}
 	if ncd.Attributes == nil {
-		ncd.Attributes = make(map[string]*attribute)
+		ncd.Attributes = make(map[AttrKey]Attribute)
 	}
-	ncd.Attributes[name] = nAttr
+	ncd.Attributes[nAttr.Key] = nAttr
```

Comment:
We are not checking if the custom attribute already exists.

I guess it is the correct behaviour, needed for update etc.?